test/Microsoft.Azure.Devices.Edge.Test.Common/linux/EdgeDaemon.cs

```diff
@@ -146,6 +149,28 @@ async Task InternalStartAsync(CancellationToken token)
             string[] output = await Process.RunAsync("systemctl", "start aziot-keyd aziot-certd aziot-identityd aziot-edged", token);
             Log.Verbose(string.Join("\n", output));
             await WaitForStatusAsync(ServiceControllerStatus.Running, token);
+
+            // Waiting for the processes to enter the "Running" state doesn't guarantee that
+            // they are fully started and ready to accept requests. Therefore, this function
+            // must wait until a request can be processed.
+            while (true)
+            {
+                var request = System.Diagnostics.Process.Start("iotedge", "list");
+
+                if (request.WaitForExit(1000))
```

Comment:
>WaitForExit [](start = 28, length = 11)

This would be true even if the process exited with an error. Should we check for non-zero exit code and retry?

Reply:
Fixed